File: 1. 단계별로_풀어보기/10~19/14. 집합과 맵/5. 10816.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

N = int(input())
cardArr = list(map(int, input().split()))
M = int(input())
queryArr = list(map(int, input().split()))

cntArr = [0] * 20000005

for card in cardArr:
  cntArr[card] += 1

for query in queryArr:
  try:
    output(f"{cntArr[query]} ")
  except:
    output(f"0 ")

File: 1. 단계별로_풀어보기/10~19/14. 집합과 맵/5. 10816.js
// 입력
const fs = require('fs')
const inputArr = fs.readFileSync(0).toString().trim().split('\n')

// 입력 파싱
const N = +inputArr[0]
const cardArr = inputArr[1].split(' ').map(Number)
const M = +inputArr[2]
const queryArr = inputArr[3].split(' ').map(Number)

// 카운팅 저장할 변수
const cntDict = {}

// 카운팅
cardArr.forEach(card => {
  if (!cntDict[card]) cntDict[card] = 1
  else cntDict[card] += 1
})

// 출력
let result = ''
queryArr.forEach(query => {
  if (!cntDict[query]) result += '0 '
  else result += `${cntDict[query]} `
})
process.stdout.write(result)
